### DISCUSSION TITLE
Collection of Reducers, Scope and item access ?

### DISCUSSION THREAD

**Discussion by JeromeTonnelierOgury** [pointfreeco/episode-code-samples#147] [Q&A]
Hi everyone,

still continuing to learn TCA, and I encoutered a weird case, I don't how to resolve this one 😓
Let's say I have a list of ingredient (flour, sugar, chocolate, ...) and I want to present a controller that allows me to choose the quantity of each ingredient.

I ordered my objects like this basically

```swift
enum IngredientType {
   case flour
   case sugar
   case chocolate
   case aubergine
   case courgette
   case butternut
   // yes, very eclectic :D :D 
}

struct Ingredient {
   let type: IngredientType
   var nbOfItems = 0
}

struct IngredientSection {
   let title: String
   var ingredients: [Ingredient]
}
```

I have a view with a stepper that allows me to increment the nb of a particular item. So I created a Reducer that holds an Ingredient as a State and it works fine.

Now I would like to put it all together with a double scroll view (an horizontal list of Ingredients ("Section") and a vertical list of sections, just like the App Store App for example).

So I have a reducer's State that will look like this 
```swift
struct State: Equatable {
   var sections: [IngredientSection]
}
```

I managed to create the view, but I'm stuck when I have to `scope` my main reducer into the small one. Usually, from what I saw and used, we have to create a target reducer State inside the parent's state and use setters and getter to do the magic. But here, I have to do this from an array, so I cannot use a variable. 
Is it an architectural issue or more likely, the geniuses of PointFree have developed a stuff I'm not aware of :D 

Thanks for your help <3

**Comment by mbrandonw** [comment]
Hi @JeromeTonnelierOgury, I think this question would be best to ask in the [library's repo](https://github.com/pointfreeco/swift-composable-architecture/discussions) so that more people can see it. And when you start a discussion over there, it would help to provide a bit more code to demonstrate the problem.

But, as just one quick comment, if the horizontal lists of ingredients do not have significant behavior in them, such as side effects, then you may not need to split up the feature into smaller pieces. You can just use a collection of values and use `ForEach` just like you would in a vanilla SwiftUI application.

**Comment by JeromeTonnelierOgury** [comment]
Ok thanks. 
My bad, I thought this was the right place. 
Should I just close it and re-open a new one where you pointed ? 
Thanks

**Reply by mbrandonw** [reply]
Yes, please open a new discussion in the other repo to get more eyes on it from the community.

**Reply by JeromeTonnelierOgury** [reply]
Thx, just did with the extra code I added below.

**Comment by JeromeTonnelierOgury** [comment]
> But, as just one quick comment, if the horizontal lists of ingredients do not have significant behavior in them, such as side effects, then you may not need to split up the feature into smaller pieces. You can just use a collection of values and use `ForEach` just like you would in a vanilla SwiftUI application.

Ok but a Stepper needs a binding, how can I use it with a Collection (Array, set, ...)

```swift
ForEach(viewStore.section) { section in 
   VStack {
      Text(section.title)
      HStack {
         ForEach(section.ingredient) ( ingredient in 
            Stepper("",
                    value: viewStore.$section[???] // <-- how can I use index here ? $section is a Binding, so $section[index] does not work
                    )
         }
      }
   }
}
```
